app/models/data_attribute.rb

```diff
@@ -37,7 +37,11 @@ class DataAttribute < ApplicationRecord
 
   belongs_to :predicate, foreign_key: 'controlled_vocabulary_term_id', class_name: 'Predicate', inverse_of: :internal_attributes
 
-  validates_presence_of :type, :value
+  validates :value, presence: true
+
+  validates :type, inclusion: {
```

Comment:
I'm not sure we need this.  Try making `DataAttribute` an abstract class and see if it does the same thing.

Reply:
OK, thanks for experimenting.